### DISCUSSION TITLE
groupby and multiple taxonomies?

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/cms#6070] [General]
Hi

Is it possible to use the new `groupby` operator to group entries by multiple categories? Or is there another way to do this?

I've got a `divisions` taxonomy, located at `resources/views/divisions/show.antlers.html`

I'm using this to output the entires from a `products` collection, when viewing a division term:

```
{{ title | wrap:h2.font-medium }}
{{ taxonomy from="divisions" collection="products" :slug:is="slug"}}
    <ul> 
        {{ entries }}
            <li><a href="{{ url }}">{{ title }}</a></li>
        {{ /entries }}
    </ul>
{{ /taxonomy }}
```

This is an example entry from the product collection:
```
---
id: 04757f84-712f-4c1a-93fa-e765b4241a65
blueprint: products
title: 'Automated Handling'
updated_by: a22581f2-580e-4041-81e2-e343df73d946
updated_at: 1653057700
p1:
  - automation-and-materials-handling
divisions: technologies-and-services
---
```
This entry has another taxonomy, called `p1`, other entries in this products collection have multiple taxonomies, `p1`, `p2`, `p3` and `p4`.

I'm trying to list the entries based on the `divisions` taxonomy and then list/group the entries by `p1`, `p2`, `p3` and `p4` terms.

I can't wrap my head around the example in the docs, how it'd be applied to this.

**Comment by stephenmeehanuk** [comment]
Looking over the docs again, I was hoping this would work...  but no joy?

```
{{ products = {collection:products} groupby (divisions, p1, p2, p3, p4) as 'entries' }}
  <h2>{{ title }}</h2>
  <ul>
    {{ entries }}
      <li><a href="{{ url }}">{{ title }}</a></li>
    {{ /entries }}
  </ul>
{{ /products }}
```
Any pointers would be appreciated.

laravel.log shows this error
```
local.ERROR: [ANTLR_006]: Unexpected stack condition encountered. Line 25 char 549. {"view":{"view":"","data":{"__env":"<pre class=sf-dump id=sf-dump-1247388347 data-indent-pad=\"  \"><span class=sf-dump-note>Illuminate\\View\\Factory</span> {<a class=sf-dump-ref href=#sf-dump-1247388347-ref2424 title=\"13 occurrences\">#424</a><samp data-depth=1 id=sf-dump-1247388347-ref2424 class=sf-dump-expanded>
```

**Comment by JohnathonKoster** [comment]
Can `p1`, `p2`, `p3`, and `p4` have multiple terms associated with the entry?

**Reply by stephenmeehanuk** [reply]
Thanks for the reply!

The taxonomies will be limited to `max_items: 1`, but an entry could have multiple taxonomies. E.g. `divisions`, `p1` and `p2`. Another entry might only have `divisions` and `p1`

**Reply by JohnathonKoster** [reply]
Fantastic! And to make sure I got this right, if entries have `divisions`, `p1` and `p2` that is a separate group from entries having `divisions` and `p1` ?

**Reply by stephenmeehanuk** [reply]
Yes, that's right.

Although thinking about it over the weekend, it needs to work slightly different from my original post.

Here's an example of an entry from `collection:products`. This entry has `p1`, `p2`, `p3` and `divisions` taxonomy terms
```
---
id: 71f258bc-cc3f-45e2-83e5-e40fdcf55620
blueprint: products
title: Waterers
updated_by: a22581f2-580e-4041-81e2-e343df73d946
updated_at: 1649347749
p1:
    - precast-concrete
p2:
    - precast-products
p3:
    - agricultural-products
divisions: precast-concrete-products
---
```

This gets me part of the way, it simply filters `collection:products` by the `division` slug.

```
{{ taxonomy from="divisions" collection="products" :slug:is="slug"}}
    <ul> 
        {{ entries }}
            <li><a href="{{ url }}">{{ title }}</a></li>
        {{ /entries }}
    </ul>
{{ /taxonomy }}
```
But I'm looking for a way to group the `entries` by taxonomies and output the terms for each taxonomy.

Is this possible?
